crates/virtual-dom-rs/src/diff/mod.rs

```diff
@@ -15,101 +16,137 @@ fn diff_recursive<'a, 'b>(
     cur_node_idx: &'b mut usize,
 ) -> Vec<Patch<'a>> {
     let mut patches = vec![];
+    let mut replace = false;
 
-    if old.tag != new.tag {
+    // Different enum variants, replace!
+    if mem::discriminant(old) != mem::discriminant(new) {
+        replace = true;
+    }
+
+    // Different element tags, replace!
+    if let (VirtualNode::Element(old_element), VirtualNode::Element(new_element)) = (old, new) {
+        if old_element.tag != new_element.tag {
+            replace = true;
+        }
+    }
+
+    // Handle replacing of a node
+    if replace {
         patches.push(Patch::Replace(*cur_node_idx, &new));
-        if let Some(children) = old.children.as_ref() {
-            for child in children {
-                increment_node_idx_for_children(child, cur_node_idx);
+        if let VirtualNode::Element(old_element_node) = old {
+            if let Some(children) = old_element_node.children.as_ref() {
+                for child in children {
+                    if let VirtualNode::Element(child_element) = child {
```

Comment:
I think that the previous behavior was incrementing the index for both text nodes as well as elements.

Was this change intentional?

I haven't looked at which tests are failing yet so just leaving this note here but if you've already addressed this on the patching side then no worries!

Reply:
Fixed in 7c1aea2. I originally overlooked that `increment_node_idx_for_children` also increments the idx even if there are no children.